imap_processing/idex/idex_packet_parser.py

```diff
@@ -0,0 +1,286 @@
+        # Extract the next 6 bits integer (usually 32)
+        hspretrigblocks = (pkt.data["IDX__TXHDRBLOCKS"].derived_value >> 6) & 0b111111
+        # Extract the first 6 bits (usually 32)
+        hsposttrigblocks = (pkt.data["IDX__TXHDRBLOCKS"].derived_value) & 0b111111
+        logging.info("HS pre trig sampling blocks: " + str(hspretrigblocks))
+        logging.info("LS pre trig sampling blocks: " + str(lspretrigblocks))
+        logging.info("HS post trig sampling blocks: " + str(hsposttrigblocks))
+        logging.info("LS post trig sampling blocks: " + str(lsposttrigblocks))
+        tof_delay = (
+            pkt.data["IDX__TXHDRSAMPDELAY"].raw_value >> 2
+        )  # First two bits are padding
+        mask = 0b1111111111
+        lgdelay = (tof_delay) & mask
+        mgdelay = (tof_delay >> 10) & mask
+        hgdelay = (tof_delay >> 20) & mask
+        logging.info(f"High gain delay = {hgdelay} samples.")
+        logging.info(f"Mid gain delay = {mgdelay} samples.")
+        logging.info(f"Low gain delay = {lgdelay} samples.")
+        if (
+            pkt.data["IDX__TXHDRLSTRIGMODE"].derived_value != 0
+        ):  # If this was a LS (Target Low Gain) trigger
+            self.Triggerorigin = "LS"
+            logging.info("Low sampling trigger mode enabled.")
+        logging.info(
+            "Packet low trigger mode = "
+            + str(pkt.data["IDX__TXHDRLGTRIGMODE"].derived_value)
+        )
+        logging.info(
+            "Packet mid trigger mode = "
+            + str(pkt.data["IDX__TXHDRMGTRIGMODE"].derived_value)
+        )
+        logging.info(
+            "Packet high trigger mode = "
+            + str(pkt.data["IDX__TXHDRHGTRIGMODE"].derived_value)
+        )
+        if pkt.data["IDX__TXHDRLGTRIGMODE"].derived_value != 0:
+            logging.info("Low gain TOF trigger mode enabled.")
+            self.Triggerorigin = "LG"
+        if pkt.data["IDX__TXHDRMGTRIGMODE"].derived_value != 0:
+            logging.info("Mid gain TOF trigger mode enabled.")
+            self.Triggerorigin = "MG"
+        if pkt.data["IDX__TXHDRHGTRIGMODE"].derived_value != 0:
+            logging.info("High gain trigger mode enabled.")
+            self.Triggerorigin = "HG"
+        logging.info(
+            f"AID = {pkt.data['IDX__SCI0AID'].derived_value}"
+        )  # Instrument event number
+        logging.info(
+            f"Event number = {pkt.data['IDX__SCI0EVTNUM'].raw_value}"
+        )  # Event number out of how many events constitute the file
+        logging.info(
+            f"Rice compression enabled = {bool(pkt.data['IDX__SCI0COMP'].raw_value)}"
+        )
+
+
+class IDEXRawDustEvent:
+    def __init__(self, epoch: float):
+        """
+        This function initializes a raw dust event
+
+        Parameters:
+            epoch (float):  Impact time in seconds since January 1st 2012 Midnight UTC
+
+        """
+        self.impact_time = epoch
+        self.TOF_High_bits = ""
```

Comment:
So these are strings in the xarray data fields as well? Or do we want to convert them to binary at some point?

Reply:
Here they are strings of just 1's and 0's read directly from the packet.  In the `process()` function they'll get converted to binary and then into integers within the _parse_high/low_sample_waveform function